phase/gather_k0s_facts.go

```diff
@@ -83,6 +85,71 @@ func (p *GatherK0sFacts) Run() error {
 		return err
 	}
 
+	isInternalEtcd := true
+	switch p.leader.Role {
+	case "single":
+		log.Debugf("%s: single node cluster, skipping etcd member list", p.leader)
+		isInternalEtcd = false
+	case "controller", "controller+worker":
+		if p.Config.Spec.K0s != nil && p.Config.Spec.K0s.Config != nil {
+			log.Debugf("%s: checking storage config for etcd", p.leader)
+			if storageConfig, ok := p.Config.Spec.K0s.Config.Dig("spec", "storage").(dig.Mapping); ok {
+				log.Debugf("%s: storage is configured", p.leader)
+				if storageConfig.DigString("type") == "kine" {
```

Comment:
Better check for type != etcd?

Reply:
Just curious, but why not always check? It seems like this could help regardless.

Reply:
I think it's about error handling I guess? If k0s doesn't use internal etcd, then `k0s etcd member-list` will fail and k0sctl would need to figure out why that is. That said, it's kinda possible I think.

String matching on stderr might be an approach that's not too bad: https://github.com/k0sproject/k0s/blob/v1.30.1%2Bk0s.0/cmd/etcd/etcd.go#L45-L50

Reply:
If k0s, storage or storageType are not configured, internal etcd should be expected.

I refactored this check into `isInternalEtcd() bool`.